Update structure and presentation

razvand opened the pull request:
# Prerequisite Checklist

<!--
Please mark items appropriately:
-->

- [x] Read the [contribution guidelines](https://github.com/open-education-hub/ccas-internal/blob/main/CONTRIBUTING.md#pull-requests) regarding submitting new changes to the project;
- [x] Tested your changes against relevant architectures and platforms;
- [x] Updated relevant documentation (if needed).

## Description of changes

<!--
Please provide a detailed description of the changes made in this new PR.
-->

Add entire contents of binary security class. Configure appropriate rendering in `config.yaml`. Make updates to content: placement, links, formatting.

razvand committed:
README.md: Update README

Make `README.md` instructions compatible with other repositories.

Signed-off-by: Razvan Deaconescu <razvan.deaconescu@upb.ro>

razvand committed:
COPYING.md: Remove trailing whitespace

Remove trailing whitespace in `COPYING.md`.

Signed-off-by: Razvan Deaconescu <razvan.deaconescu@upb.ro>

razvand committed:
CONTRIBUTING: Sync with template

Sync `CONTRIBUTING.md` contents with the one in the template repository.

Signed-off-by: Razvan Deaconescu <razvan.deaconescu@upb.ro>

razvand committed:
config.yaml: Update links

Use correct links (URLs) to binary security content (SSS) in config.yaml
file.

Signed-off-by: Razvan Deaconescu <razvan.deaconescu@upb.ro>

razvand committed:
chapters: Remove generated content

Generated content (mostly slides) was added by mistake in the
repository. Remove it.

Signed-off-by: Razvan Deaconescu <razvan.deaconescu@upb.ro>

razvand committed:
chapters: Ignore generated slide files

Add `.gitignore` file to ignore generated slide output files. Namely the
`_site/` directory and the `slides.md` file (resulted from
preprocessing).

Signed-off-by: Razvan Deaconescu <razvan.deaconescu@upb.ro>

razvand committed:
chapters: Move activities to drills, assets to media

Rename `activities/` directory to `drills/`. And rename `assets/` to
`media/`.

Also place `README.md` in `reading/` directory.

Signed-off-by: Razvan Deaconescu <razvan.deaconescu@upb.ro>

razvand committed:
chapter: Reformat content

Reformat content to make it publishable via Docusaurus.

The `pwntools-intro` section was previously stored outside the
`chapters/` directory. Move it inside the `chapters/` directory.

Update formatting, links and configuration file (`config.yaml`) to new
structure.

Reformat Python code using `black`.

Apply linting (via `markdownlint-cli`) to Markdown files.

Signed-off-by: Razvan Deaconescu <razvan.deaconescu@upb.ro>

gabrielmocanu commented:
I added the `pr-deployment` workflow to be able to render this PR.
Please update your branch to `main`.

gabrielmocanu commented:
It looks ok to me, I think we should ignore for now `SuperLinter` workflow since it will be a lot of changes there.
I would focus on rendering the website and solving `Checkpatch` and `Spellcheck` workflows.

razvand committed:
Add CPPLINT.cfg

The `CPPLINT.cfg` file in the actions repository doesn't work. The
configuration file has to be part of the root repository, so we add it
here.

Signed-off-by: Razvan Deaconescu <razvan.deaconescu@upb.ro>

teodutu reviewed (approved):
I scrolled through the content and couldn't find any rendering issues. It looks alright.
